components/script_traits/script_msg.rs

```diff
@@ -79,7 +79,8 @@ pub enum ScriptMsg {
     ChangeRunningAnimationsState(AnimationState),
     /// Requests that a new 2D canvas thread be created. (This is done in the constellation because
     /// 2D canvases may use the GPU and we don't want to give untrusted content access to the GPU.)
-    CreateCanvasPaintThread(Size2D<i32>, IpcSender<IpcSender<CanvasMsg>>),
+    CreateCanvasPaintThread(
+        Size2D<i32>, IpcSender<IpcSender<CanvasMsg>>, (IpcSender<CanvasMsg>, IpcReceiver<CanvasMsg>)),
```

Comment:
We should be able to remove the `IpcSender<IpcSender>` argument and only pass the `IpcReceiver` argument instead.